### ISSUE TITLE
Hitting "Cancel" on SignOut system popup returns a generic error.

### ISSUE THREAD

**Issue by gevartosky** [aws-amplify/amplify-ios#967]
**Describe the bug**
When trying to logout, if the user hits "Cancel" on the system popup, the callback `Amplify.Auth.signOut` fails with a generic error.

**To Reproduce**
1. Call `Amplify.Auth.signOut`
2. Hit "Cancel"
3. See a generic error retuned on the callback

**Expected behavior**
A more descriptive error would help disambiguate if the user selected cancel and would avoid not only avoid writing code to guess if the error was caused by hitting Cancel or an actual error, but we can avoid executing functions that should be executed on a real error scenario.

**Screenshots**
![image](https://user-images.githubusercontent.com/5980943/101927375-30de6c80-3bb3-11eb-9a40-0715517020db.png)

**Environment(please complete the following information):**
 - Amplify Framework Version: 1.4.1
 - Dependency Manager: Cocoapods
 - Swift Version : 5.2

**Device Information (please complete the following information):**
 - Device: Simulator
 - iOS Version: iOS 12+

**Comment by palpatim** [comment]
Yes, assuming we get an appropriate callback, we should propagate a meaningful error.

**Comment by royjit** [comment]
Hi @gevartosky 

Thank you for reporting the issue. We have created a PR to fix the issue here - https://github.com/aws-amplify/amplify-ios/pull/982

**Comment by royjit** [comment]
We are holding the PR off for now because the change will be a breaking change. For now you can check if the user cancelled the signIn or signOut flow by following the below code:

```swift

        Amplify.Auth.signInWithWebUI(presentationAnchor: self.view.window!) { (result) in
            
            switch result {
            case .success(let result):
                print(result)
            case .failure(let error):
                if case AuthError.unknown(_, let underlyingError) = error,
                   case .canceledLogin = (underlyingError as? SFAuthenticationError)?.code {
                       print("User cancelled")
                }
            }
        }

         // SignOut
        Amplify.Auth.signOut(listener: { (result) in
            
            switch result {
            case .success:
                print("SignOut")
            case .failure(let error):
                if case AuthError.unknown(_, let underlyingError) = error,
                   case .canceledLogin = (underlyingError as? SFAuthenticationError)?.code {
                       print("User cancelled")
                }
            }
        })
```

**Comment by royjit** [comment]
We have released the PR in 1.6.0 https://github.com/aws-amplify/amplify-ios/releases/tag/v1.6.0. Please take a look, and reach back if you have any issue. 

```swift
Amplify.Auth.signInWithWebUI(presentationAnchor: self.view.window!) { (result) in
            
            switch result {
            case .success(let result):
                print(result)
            case .failure(let error):
                if case AuthError.service(_, _, let underlyingError) = error,
                   case .userCancelled = (underlyingError as? AWSCognitoAuthError) {
                    print("User cancelled")
                }
            }
        }

// SignOut

Amplify.Auth.signOut(listener: { (result) in
            
            switch result {
            case .success:
                print("SignOut")
            case .failure(let error):
                if case AuthError.service(_, _, let underlyingError) = error,
                   case .userCancelled = (underlyingError as? AWSCognitoAuthError) {
                    print("User cancelled")
                }
            }
        })
```
